lib/job.js

```diff
@@ -206,6 +206,34 @@ class Job {
       });
   }
 
+  /**
+   * Heartbeat as needed, until a promise resolves.
+   *
+   * Before the job lock is lost, the heartbeat is run. The padding controls how long
+   * (in seconds) before the job lock loss to run the heartbeat. The default value is
+   * 10, meaning that the heartbeat will run 10 seconds before the lock is set to
+   * expire.
+   */
+  heartbeatUntilPromise(promise, padding) {
+    // The promise is wrapped in Bluebird to ensure it adheres to the expected API
+    const wrappedPromise = Promise.resolve(promise);
+    // Default to 10 seconds
+    padding = padding || 10;
```

Comment:
You may be able to use default parameters (I'm not 100% sure at which node version this was supported, I'm on node 6.10)

```
$ node
> function x(foo = 5) { console.log(foo); }
undefined
> x()
5
undefined
```